docs/extending.rst

```diff
@@ -71,59 +75,78 @@ The above schema has a member named ``interested`` which will now be
-which expects a schema node as its first constructor argument.  ``cstruct``
-will be the :term:`cstruct` value that needs to be deserialized.
+Arguments:
+
+- ``node``: the ``SchemaNode`` associated with this type
+- ``cstruct``: the :term:`cstruct` that the caller wants to obtain child values
+  for
+
+You only need to define this method for complex types that have child nodes,
+such as mappings and sequences.
+
+``cstruct_children`` should return a value based on ``cstruct`` for
+each child node in ``node`` (or an empty list if ``node`` has no children). If
+``cstruct`` does not contain a value for a particular child, that child should
+be replaced with the ``colander.null`` value in the returned list.
+
+``cstruct_children`` should *never* raise an exception, even if it is passed a
+nonsensical ``cstruct`` argument. In that case, it should return a sequence of
+as many ``colander.null`` values as there are child nodes.
+
+
+Constructor (``__init__``)
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+`SchemaType` does not define a constructor, and user code (not Colander)
+instantiates type objects, so custom types may define this method and use it
+for their own purposes.
 
-The ``cstruct_children`` method accepts two values: ``node`` and ``cstruct``.
-``node`` will be the schema node associated with this type.  ``cstruct`` will
-be the :term:`cstruct` that the caller wants to obtain child values for.  The
-``cstruct_children`` method should *never* raise an exception, even if it
-passed a nonsensical value.  If it is passed a nonsensical value, it should
-return a sequence of ``colander.null`` values; the sequence should contain as
-many nulls as there are node children.  If the ``cstruct`` passed does not
-contain a value for a particular child, that child should be replaced with
-the ``colander.null`` value in the returned list.  Generally, if the type
-you're defining is not expected to have children, it's fine to return an
-empty list from ``cstruct_children``.  It's only useful for complex types
-such as mappings and sequences, usually.
 
 Null Values
 ~~~~~~~~~~~
 
-The framework requires that both the ``serialize`` method and the
-``deserialize`` method of a type explicitly deal with the potential to
-receive a :attr:`colander.null` value.  :attr:`colander.null` will be sent to
-the type during serialization and deserialization in circumstances where a
-value has not been provided by the data structure being serialized or
-deserialized.  In the common case, when the ``serialize`` or ``deserialize``
-method of type receives the :attr:`colander.null` value, it should just
-return :attr:`colander.null` to its caller.
+Both the ``serialize`` and ``deserialize`` methods must be able to
+receive :attr:`colander.null` values and handle them intelligently. This
+will happen whenever the data structure being serialized or deserialized
+does not provide a value for this node. In most cases, ``serialize`` or
```

Comment:
"In the common case" is deliberately weaker than "in most cases":  perhaps "in many cases" instead?